cmd/run.go

```diff
@@ -113,23 +116,30 @@ func NewRunner(config RunConfig) *Runner {
 // Run runs all matching features for the given patterns (or all if no patterns
 // given).
 func (r *Runner) Run(ctx context.Context, patterns []string) error {
-	switch r.config.Lang {
-	case "go", "java", "ts", "py":
-		// Allow the full typescript or python word, but we need to match the file
-		// extension for the rest of run
-	case "typescript":
-		r.config.Lang = "ts"
-	case "python":
-		r.config.Lang = "py"
-	default:
-		return fmt.Errorf("invalid language %q, must be one of: go or java or ts or py", r.config.Lang)
+	var err error
+	if r.config.Lang, err = normalizeLangName(r.config.Lang); err != nil {
+		return err
 	}
 
 	// Cannot generate history if a version isn't provided explicitly
 	if r.config.GenerateHistory && r.config.Version == "" {
 		return fmt.Errorf("must have explicit version to generate history")
 	}
 
+	// If prepared dir given, validate and make absolute
+	if r.config.Dir != "" {
+		if strings.ContainsAny(r.config.Dir, `\/`) {
+			return fmt.Errorf("prepared directory must not have path separators, it is always relative to the SDK features root")
```

Comment:
No subdirectories?

Reply:
No. Right now all of the harnesses are expected to be in `../harness`. We can fix that later, but for now the temporary directories and the prepared directories are expected to be just beneath the sdk-features directory.